docs/src/modeler/time_series.md

```diff
@@ -0,0 +1,319 @@
+# Managing Time Series Data
+
+PowerSystems supports two categories of time series data depending on the
+process to obtain the data:
+
+- Static data: a single column of time series values for a component field
+(such as active power) where each time period is represented by a single value.
+This data commonly is obtained from historical information or the realization of
+a time-varying quantity.
+- Forecasts: Predicted values of a time-varying quantity that commonly features
+a look-ahead and can have multiple data values representing each time period.
+This data is used in simulation with receding horizons or data generated from
+forecasting algorithms.
+
+## Types
+
+### Static Time Series Data
+
+PowerSystems defines the Julia struct `SingleTimeSeries` to represent this data.
+
+### Forecasts
+
+PowerSystems defines the following Julia structs to represent forecasts:
+
+- `Deterministic`: Point forecast without any uncertainty representation.
+- `Probabilistic`: Stores a discretized cumulative distribution functions
+(CDFs) or probability distribution functions (PDFs) at each time step in the
+look-ahead window.
+- `Scenarios`: Stores a set of probable trajectories for forecasted quantity
+with equal probability.
+
+## Storage
+
+By default PowerSystems stores time series data in an HDF5 file. This prevents
+large datasets from overwhelming system memory. If you know that your dataset
+will fit in your computer's memory then you can increase performance by storing
+it in memory. Here is an example of how to do this:
+
+```julia
+sys = System(100.0; time_series_in_memory = true)
+```
+
+## Creating Time Series Data
+
+PowerSystems supports several methods to load time series data (Forecasts or
+StaticTimeSeries) into a System.
+
+- Automated parsing during system construction. Refer to the
+[parsing documentation](parsing.md).
+- Create from TimeSeries.TimeArray or DataFrames.DataFrame
+
+```julia
+    resolution = Dates.Hour(1)
+    dates = range(DateTime("2020-01-01T00:00:00"), step = resolution, length = 24)
+    data = TimeArray(dates, ones(24))
+    ts = SingleTimeSeries("max_active_power", data)
+```
+
+```julia
```

Comment:
I am familiar with the `TimeArray` approach, but I was not with this one. It's nice to know that there is an alternative that doesn't necessarily depend on `TimeSeries.jl`.

However, I'm not sure the example is clear to me. The example above defines a `TimeArray` from `2020-01-01T00:00:00` to `2020-01-01T23:00:00` and assigns `ones(24)` to it, which means for each hour the forecast is `1.0`.

In the example below, we have `ones(24)` associated to each of the two hours defined, which seems very different. I think maybe for clarity it would be better if both examples are perfectly equivalent? I apologize if they are and I'm missing something.

Reply:
@raphaelsaavedra you are correct, they aren't equivalent. The first one creates a `SingleTimeSeries` object, and the second one creates a `Deterministic` object with two 24hr forecast windows. I agree that it's confusing to create two different types of objects here. I think the point was to demonstrate that both types of objects could be created.
We should either add a description of what's going on, align the two examples, or provide four examples to enumerate all the options.

Reply:
i think this is something we've found a little confusing, so think just giving all 4 examples might be worth it.

Reply:
We are also introducing a completely new time series formatting so there will be additional interfaces.

Reply:
Thanks for the answer!

> the second one creates a Deterministic object with two 24hr forecast windows.

That's the root of my confusion (I'm just trying to understand this second approach). In the `TimeArray` approach, we use a `TimeArray` that explicitly specifies the hours to be considered in the forecast (via `range`) so that each hour is associated to a float (the range `2020-01-01T00:00:00:1 hour:2020-01-01T23:00:00` is associated with `ones(24)` in this example).

Conversely, in the example below, the `DateTime` `range` is never specified:

```
resolution = Dates.Hour(1)
data = Dict(
    DateTime("2020-01-01T00:00:00") => ones(24),
    DateTime("2020-01-01T01:00:00") => ones(24),
)
forecast = Deterministic("max_active_power", data, resolution)
```

What we have two `DateTime`s, each pointing to a forecast of `ones(24)`. Is this supposed to indicate "a forecast (with hourly `resolution`) starting from `DateTime("2020-01-01T00:00:00")` and with values `ones(24)`? That doesn't seem the case because that would mean the second forecast starts one hour after the first one.

Well, I guess in summary I think the example could use some further clarification, because it's not obvious to me what's happening.

Reply:
@raphaelsaavedra the content in 42eda80 should help clarify this questions.

Reply:
> the key is the initial time of the forecast and the value
field is the forecast value

Thanks! That answers my question. My last question is: doesn't the fact that they start in different instants go against the time series validation?

> All forecasts must have identical parameters:  initial timestamp, horizon, interval, look-ahead window count.

Reply:
What do you mean? "doesn't the fact that they start in different instants go against the time series validation?"

Reply:
Maybe I am still misunderstanding the code. There are two time series in that forecast, one is `10.0*ones(24)` starting at `2020-01-01T00:00:00` and the other is `5.0*ones(24)` starting at `2020-01-01T01:00:00`. They start at different instants, so wouldn't that cause an error when validating the time series?

Reply:
I would state this a bit differently.  In the code a `forecast` is a collection of look-ahead windows.  Each forecast has one initial timestamp.  Each window in the forecast will have an initial time that increments by `interval`.  Each window is an entry in the dict above.

All `forecasts` in the system must have the same initial timestamp and other parameters.

Reply:
Thanks! I think I understand it now :)